### PR TITLE
Fix/inventory

### PR DESCRIPTION
**Pull request by siddhantk232**
{Fixes design issues,
Uses new components (Form v2, Filler, etc.),
error handling using error boundary and sentry logging} in the inventory app.

edit 1: Most of the components which can be replaced by `<Flex />` has been replaced now. There may be some places which I have missed but they will be replaced in future revisions

edit 2: This PR has gone quite big now. Future changes will be done in small PRs :peace_symbol: 

##### This PR has:

-  [x] Commit messages that are correctly formatted
-  [x] Feature list updated for newly introduced code

This PR is a feature.

**Developers**

@siddhantk232

### CODE REVIEW

File: src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js [hunk 1]
```diff
@@ -37,20 +93,38 @@ export default function ProcessingTunnel({ close, open, formState }) {
...
+                  close={closeConfigTunnel}
+                  open={openConfigTunnel}
+                  proc={{}}
+                  fromTunnel
+                  closeParent={() => close(1)}
+                  id={bulkItems[0]?.id}
+               />
+            </Tunnel>
+         </Tunnels>
          <TunnelHeader
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js:122]: shouldnt this `Tunnels` be declared outside where `Processing` tunnel is declared?

File: src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+   const {
+      loading: itemsLoading,
+      data: { bulkItems = [] } = {},
+      error,
+   } = useSubscription(SUPPLIER_ITEMS_LISTINGS_BULK)
+   const { tooltip } = useTooltip()
+
+   if (error) {
+      logger(error)
+      throw error // let this error catched by the ErrorBoundary as the view requires this.
```
Comment (prvnbist) [inline src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx:22]: why is this here?

File: src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      autoResize: true,
+      maxHeight: '420px',
+      resizableColumns: false,
+      virtualDomBuffer: 80,
+      placeholder: 'No Data Available',
+      persistence: true,
+      persistenceMode: 'cookie',
+      pagination: 'local',
+      paginationSize: 10,
+   }
```
Comment (prvnbist) [inline src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx:37]: When we're keeping the table options in root of the app, why is this here then?

File: src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+            <ReactTabulator
+               ref={tableRef}
+               columns={columns}
+               data={bulkItems}
+               options={{
+                  ...tableOptions,
+                  groupBy: 'supplierItem.name',
+                  selectable: true,
+               }}
+               style={{ marginTop: '16px' }}
```
Comment (prvnbist) [inline src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx:156]: Use `Spacer` component instead

File: src/apps/inventory/views/Listings/Item/index.js [hunk 5]
```diff
@@ -46,123 +44,48 @@ export default function ItemListing() {
...
-      resizableColumns: false,
-      virtualDomBuffer: 80,
-      placeholder: 'No Data Available',
-      persistence: true,
-      persistenceMode: 'cookie',
-      pagination: 'local',
-      paginationSize: 10,
-   }
 
    const tableRef = React.useRef()
```
Comment (prvnbist) [inline src/apps/inventory/views/Listings/Item/index.js:48]: Shouldn't there be two `tableRef` for each supplier item table and bulk item listing?

File: src/apps/inventory/views/Listings/PurchaseOrders/SelectPurchaseOrderTypeTunnel.jsx [hunk 6]
```diff
@@ -54,31 +60,26 @@ export default function SelectPurchaseOrderTypeTunnel({ close }) {
...
-            <Text as="subtitle">
-               Purchase orders associated with inventory supplier items.
-            </Text>
-         </SolidTile>
-      </TunnelContainer>
+         <TunnelContainer>
+            <SolidTile onClick={createPackagingPurchaseOrder}>
+               <Text as="h1">Packaging</Text>
+               <Text as="subtitle">
+                  {PACKAGING_PURCHASE_ORDERS_CREATE_TUNNEL}
```
Comment (prvnbist) [inline src/apps/inventory/views/Listings/PurchaseOrders/SelectPurchaseOrderTypeTunnel.jsx:74]: why was replacing the plain text with variable necessary?

File: src/apps/inventory/views/Listings/PurchaseOrders/index.jsx [hunk 7]
```diff
@@ -91,21 +119,19 @@ export default function PurchaseOrders() {
...
-            <div style={{ width: '90%', margin: '0 auto' }}>
-               <ReactTabulator
-                  ref={tableRef}
-                  columns={columns}
-                  data={purchaseOrderItems}
-                  rowClick={rowClick}
-                  options={tableOptions}
-               />
-            </div>
+            <br />
```
Comment (prvnbist) [inline src/apps/inventory/views/Listings/PurchaseOrders/index.jsx:128]: Use `Spacer`

File: src/apps/inventory/views/Listings/PurchaseOrders/index.jsx [hunk 8]
```diff
@@ -76,13 +101,16 @@ export default function PurchaseOrders() {
    return (
       <>
          <Tunnels tunnels={tunnels}>
-            <Tunnel layer={1}>
+            <Tunnel layer={1} size="sm">
                <SelectPurchaseOrderTypeTunnel close={closeTunnel} />
             </Tunnel>
          </Tunnels>
          <StyledWrapper>
```
Comment (prvnbist) [inline src/apps/inventory/views/Listings/PurchaseOrders/index.jsx:108]: Use `Flex`

File: src/apps/inventory/views/Listings/WorkOrders/WorkOrderTypeTunnel.jsx [hunk 9]
```diff
@@ -52,37 +56,32 @@ export default function WorkOrderTypeTunnel({ close }) {
...
-               </Trans>
-            </Text>
-         </SolidTile>
-      </TunnelContainer>
+         <TunnelContainer>
+            <SolidTile onClick={createBulkWorkOrder}>
+               <Text as="h1">{t(address.concat('bulk work order'))}</Text>
+               <Text as="subtitle">
+                  <Trans i18nKey={address.concat('bulk subtitle 1')}>
+                     {BULK_WORK_ORDER_INFO_TUNNEL}
```
Comment (prvnbist) [inline src/apps/inventory/views/Listings/WorkOrders/WorkOrderTypeTunnel.jsx:71]: Why is this variable necessary when it could've been just a text

File: src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js (hunk 1 above)
Comment (siddhantk232) [reply src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js:122]: for editing, this tunnel needs information about the selected bulk item which is possible here only.

File: src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx (hunk 2 above)
Comment (siddhantk232) [reply src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx:22]: omitting this will also let the error boundary to catch this error. This is just to log the error to sentry.

File: src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx (hunk 3 above)
Comment (siddhantk232) [reply src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx:37]: I may have missed this one, will change it today.

File: src/apps/inventory/views/Listings/Item/index.js (hunk 5 above)
Comment (siddhantk232) [reply src/apps/inventory/views/Listings/Item/index.js:48]: the clear filters button is in the parent listings view. The tables are using ref from parent to use the same clear button to clear the filters applied to the active table

File: src/apps/inventory/views/Listings/PurchaseOrders/SelectPurchaseOrderTypeTunnel.jsx (hunk 6 above)
Comment (siddhantk232) [reply src/apps/inventory/views/Listings/PurchaseOrders/SelectPurchaseOrderTypeTunnel.jsx:74]: I don't know why I did that. will change that today

File: src/apps/inventory/views/Listings/PurchaseOrders/index.jsx (hunk 8 above)
Comment (siddhantk232) [reply src/apps/inventory/views/Listings/PurchaseOrders/index.jsx:108]: where?

File: src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js (hunk 1 above)
Comment (prvnbist) [reply src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js:122]: you move the data to closest parent if both of them requires it, you dont move the component, is that the case?

File: src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx (hunk 2 above)
Comment (prvnbist) [reply src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx:22]: i'm talking about `throw error`?

File: src/apps/inventory/views/Listings/PurchaseOrders/index.jsx (hunk 8 above)
Comment (prvnbist) [reply src/apps/inventory/views/Listings/PurchaseOrders/index.jsx:108]: `StyledWrapper`

File: src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js (hunk 1 above)
Comment (siddhantk232) [reply src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js:122]: if this ConfigTunnel is declared outside, I cannot pass the id of the newly created bulkItem in this tunnel.

File: src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx (hunk 2 above)
Comment (siddhantk232) [reply src/apps/inventory/views/Listings/Item/bulkItemsListing.jsx:22]: to crash the app, so that the error boundary catches it

File: src/apps/inventory/views/Forms/Item/ProcessingView.jsx [hunk 10]
```diff
@@ -1,102 +1,173 @@
...
+            <HorizontalTabPanels>
+               <HorizontalTabPanel>
+                  {proc ? (
+                     <Flex container justifyContent="flex-end">
+                        <IconButton
+                           onClick={() =>
+                              isDefault
+                                 ? openConfigTunnel(1)
+                                 : openConfigTunnel(1)
+                           }
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/Item/ProcessingView.jsx:92]: is this a typo?

File: src/apps/inventory/views/Forms/Item/ProcessingView.jsx [hunk 11]
```diff
@@ -1,102 +1,173 @@
...
 
-         {activeView === 'plannedLot' && (
-            <>
-               <PlannedLotView open={open} formState={formState} />
-            </>
-         )}
-      </>
+   return (
+      <FlexContainer>
+         <Flexible width="4">
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/Item/ProcessingView.jsx:131]: Replace these with `Flex` component everywhere

File: src/apps/inventory/views/Forms/Item/styled.js [hunk 12]
```diff
@@ -6,9 +6,7 @@ export const Container = styled.div`
 `
 
 export const StyledHeader = styled.div`
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/Item/styled.js:8]: This can be replaced with `<Flex as="header">`, replace everywhere

File: src/apps/inventory/views/Forms/Item/styled.js [hunk 13]
```diff
@@ -152,30 +148,30 @@ export const StyledGrid = styled.div`
...
-                  font-size: 16px;
-                  line-height: 19px;
-               }
             }
          }
       }
    }
 `
+
+export const StatHeader = styled.div`
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/Item/styled.js:163]: Use `Text` component

File: src/apps/inventory/views/Forms/Item/tunnels/Allergens/index.js [hunk 14]
```diff
@@ -12,35 +12,52 @@ import {
...
+            object: {
+               allergens: selected,
+            },
          },
       })
-      close()
    }
 
-   if (allergensLoading) return <Loader />
+   if (allergensLoading || loading) return <Loader />
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/Item/tunnels/Allergens/index.js:60]: Use `<InlineLoader />` component

File: src/apps/inventory/views/Forms/Item/tunnels/Info/index.js [hunk 15]
```diff
@@ -94,173 +88,125 @@ export default function InfoTunnel({ close, formState }) {
...
-                              })
-                           }
-                        >
-                           <option value="cs">{t('units.cs')}</option>
-                        </StyledSelect>
-                     </InputWrapper>
-                  </Highlight>
-               </StyledInputGroup>
-            </StyledRow> */}
             <StyledRow>
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/Item/tunnels/Info/index.js:174]: Can be replaced with `Spacer`, since it only has `margin-bottom`

File: src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js [hunk 16]
```diff
@@ -7,23 +7,79 @@ import {
...
+   const [configTunnel, openConfigTunnel, closeConfigTunnel] = useTunnel(1)
+
+   const [updateSupplierItem] = useMutation(UPDATE_SUPPLIER_ITEM, {
+      onCompleted: () => {
+         toast.info(BULK_ITEM_CREATED)
+         toast.success(BULK_ITEM_AS_SHIPPED_ADDED)
+         openConfigTunnel(1)
+      },
+      onError: error => {
+         console.log(error)
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js:49]: Missing `logger`

File: src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js [hunk 17]
```diff
@@ -7,23 +7,79 @@ import {
...
+   })
+
+   const [
+      createBulkItem,
+      {
+         loading,
+         data: { createBulkItem: { returning: bulkItems = [] } = {} } = {},
+      },
+   ] = useMutation(CREATE_BULK_ITEM, {
+      onCompleted: data => {
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/Item/tunnels/Processing/index.js:62]: Missing `toast`

File: src/apps/inventory/views/Forms/PackagingPurchaseOrders/index.jsx [hunk 18]
```diff
@@ -63,41 +71,36 @@ export default function PackagingPurchaseOrderForm() {
...
-            </FormHeading>
-
-            <StyledForm style={{ width: '90%', margin: '0 auto' }}>
-               <Content
-                  item={item}
-                  orderQuantity={orderQuantity}
-                  setOrderQuantity={setOrderQuantity}
-               />
-            </StyledForm>
+            <StyledHeader>
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/PackagingPurchaseOrders/index.jsx:84]: Can be replaced w/ `Flex`, whichever styled component can be replaced with `Flex`, please do

File: src/apps/inventory/views/Forms/PurchaseOrders/index.jsx [hunk 19]
```diff
@@ -92,99 +106,79 @@ export default function PurchaseOrderForm() {
...
+                        onHand={state.supplierItem?.bulkItemAsShipped?.onHand}
+                        edit={() => openTunnel(1)}
+                     />
+                  )}
+                  <Separator />
+
+                  <Flex container alignItems="flex-end">
+                     <Form.Group>
+                        <Form.Label htmlFor="quantity" title="quantity">
+                           {t(address.concat('enter order quantity'))}
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/PurchaseOrders/index.jsx:135]: Missing `Tooltip`, add them to every form label

File: src/apps/inventory/views/Forms/SachetPackaging/PackagingStatus.jsx [hunk 20]
```diff
@@ -45,9 +50,9 @@ export default function PackagingStats({ state }) {
                      onClick={() => openPhotoTunnel(1)}
                   />
                )}
-            </Flexible>
-            <span style={{ width: '20px' }} />
-            <Flexible width="3">
+            </Flex>
+            <Spacer xAxis size="16px" />
+            <Flex style={{ flex: 3 }}>
```
Comment (prvnbist) [inline src/apps/inventory/views/Forms/SachetPackaging/PackagingStatus.jsx:55]: Flex component has a prop `flex`
